contracts/andromeda_primitive/src/contract.rs

```diff
@@ -89,10 +113,34 @@ pub fn execute_delete_value(
 }
 
 #[cfg_attr(not(feature = "library"), entry_point)]
-pub fn query(deps: Deps, _env: Env, msg: QueryMsg) -> StdResult<Binary> {
+pub fn query(deps: Deps, env: Env, msg: QueryMsg) -> StdResult<Binary> {
     match msg {
+        QueryMsg::AndrQuery(msg) => handle_andromeda_query(deps, env, msg),
         QueryMsg::GetValue { name } => to_binary(&query_value(deps, name)?),
-        QueryMsg::ContractOwner {} => to_binary(&query_contract_owner(deps)?),
+    }
+}
+
+fn handle_andromeda_query(deps: Deps, env: Env, msg: AndromedaQuery) -> StdResult<Binary> {
+    match msg {
+        AndromedaQuery::Get(data) => {
+            let received: Result<QueryMsg, ContractError> = parse_message(data);
+            if received.is_err() {
+                return Err(StdError::ParseErr {
```

Comment:
I don't really like this at all but I dont see an alternative at the moment. Do you think that we should just replace all `StdResult` in queries with `Result`?

Reply:
You don't like using `StdResult` specifically? I think using `Result` should be fine